src/auth-service/utils/join.js

```diff
@@ -132,8 +132,26 @@ const join = {
           const indexOfCode = indexBeforeCode + 1;
           let emailLinkCode = linkSegments[indexOfCode].substring(2);
 
-          const token = Math.floor(Math.random() * (999999 - 100000) + 100000);
+          await validationsUtil.checkEmailExistenceUsingKickbox(
+            email,
+            (value) => {
+              if (value.success == false) {
+                const errors = value.errors ? value.errors : "";
+                callback({
+                  success: false,
+                  message: value.message,
+                  errors,
+                  status: value.status,
+                });
+              }
+            }
+          );
+
           let responseFromSendEmail = {};
+          let token = 100000;
+          if (email !== "airqo.analytics@gmail.com") {
```

Comment:
@Baalmart the functionality works as expected. Considering security and easy of change, could we store the email in an environmental variable other than hard coding

Reply:
True story! Shall embark on that asap. @NoahNsimbe